src/components/Zoom/Zoom.js

```diff
@@ -91,6 +95,37 @@ function Zoom({
   );
 
   useEffect(() => {
+    const view = map.getView();
+    /* Some apps (e.g. Trafimage-Maps) do not load the correct zoom in time
+     * to update the "disabledControls" object, so we trigger a seperate hook (see below)
+     * using setZoom & currentZoom, which ensures it is written on mount and update.
+     */
+    setZoom(view.getZoom());
+    view.on('propertychange', () => {
+      /* Add view listener to trigger zoom update on view change. */
+      setZoom(view.getZoom());
+    });
+  }, []);
+
+  useEffect(() => {
+    const view = map.getView();
+    // We use the constrained max and min zoom, which is calculated including the viewport size
+    setDisabledControls({
+      zoomIn: view.getZoom() >= view.getConstrainedZoom(view.getMaxZoom()),
+      zoomOut: view.getZoom() <= view.getConstrainedZoom(view.getMinZoom()),
+    });
+  }, [currentZoom]);
```

Comment:
Can you try with useMEmo:

```
const zoomInDisabled = useMemo(()=> {
return view.getZoom() >= view.getConstrainedZoom(view.getMaxZoom());
}, [view.getZoom()])
```
I don't know if it's work, but itcould save some lines of code

Reply:
Done